### DISCUSSION TITLE
`{instance.uuid:x}{ext}` raises an error

### DISCUSSION THREAD

**Discussion by amureki** [codingjoe/django-dynamic-filenames#78] [Ideas]
Greetings there, many thanks for the package! 👍 

I have a usecase that is currently not supported:

```python
class Item(models.Model):
    uid = models.UUIDField(
        default=uuid.uuid4,
        unique=True,
    ) # 3d678d69-44f1-4015-93fa-e3426eaca4fe
    image = models.ImageField(
        upload_to=FilePattern(
            filename_pattern=(
                "{instance.uuid:x}{ext}"
            ),
        ),
    )
    # image should be stored as hex version of a given uuid
    #  3d678d6944f1401593fae3426eaca4fe.jpg 
```

I could imagine extending [`SlugFormatter`](https://github.com/codingjoe/django-dynamic-filenames/blob/main/dynamic_filenames.py#L10-L19) (and also renaming it to fit an extended purpose) to support `uuid` type.
Here is a dirty example to give an idea:

```python
class SlugFormatter(Formatter):
    format_spec_pattern = re.compile(r"(\.\d+)?([\d\w]+)?")

    def format_field(self, value, format_spec):
        precision, ftype = self.format_spec_pattern.match(format_spec).groups()
        if precision:
            precision = int(precision.lstrip("."))
        if ftype == "slug":
            return slugify(value, allow_unicode=False)[:precision]
        if ftype == "uuid":
            return value.hex[:precision]
        return super().format_field(value=value, format_spec=format_spec)
```

But I am not sure if you'd be willing to go this direction.
Happy to hear your opinion here!

Best,
Rust

**Comment by codingjoe** [comment]
Hi there,

Sadly, Python's UUID implementation doesn't support more comprehensive `__format__` options.
This package implements its own UUID version, could cast UUIDs in the formatter.
However, you probably shouldn't use the `type` descriptor. As that once should describe the output, not the input type.

You are welcome to open a PR with a suggestion.
Cheers!
Joe
